llvm/lib/Target/AMDGPU/Utils/AMDGPUBaseInfo.cpp

```diff
@@ -657,9 +658,14 @@ bool isTrue16Inst(unsigned Opc) {
   return Info ? Info->IsTrue16 : false;
 }
 
-bool isFP8DstSelInst(unsigned Opc) {
-  const FP8DstByteSelInfo *Info = getFP8DstByteSelHelper(Opc);
-  return Info ? Info->HasFP8DstByteSel : false;
+FPType getFPDstSelType(unsigned Opc) {
+  const FP4FP8DstByteSelInfo *Info = getFP4FP8DstByteSelHelper(Opc);
+  if (Info && Info->HasFP8DstByteSel)
+    return FPType::FP8;
+  else if (Info && Info->HasFP4DstByteSel)
+    return FPType::FP4;
+  else
+    return FPType::None;
```

Comment:
```suggestion
  if (!Info)
    return FPType::None;
  if (Info->HasFP8DstByteSel)
    return FPType::FP8;
  if (Info->HasFP4DstByteSel)
    return FPType::FP4;

  llvm_unreachable("not fp4 or fp8");
```
No else after return

Reply:
Need to use `return None` instead of `llvm_unreachable`. We can reach here for other instructions for handling dst forwarding hazard.